### PR TITLE
[ISSUE #4776]Support tag in litepullconsumer assign mode

### PR DESCRIPTION
**Pull request by lizhiboo**

### CODE REVIEW

File: client/src/main/java/org/apache/rocketmq/client/consumer/DefaultLitePullConsumer.java [hunk 1]
```diff
@@ -271,6 +271,11 @@ public void assign(Collection<MessageQueue> messageQueues) {
         defaultLitePullConsumerImpl.assign(queuesWithNamespace(messageQueues));
     }
 
+    @Override
+    public void setSubExpression4Assgin(final String topic, final String subExpresion) {
+        defaultLitePullConsumerImpl.setSubExpression4Assgin(topic, subExpresion);
```
Comment (ShannonDing) [inline client/src/main/java/org/apache/rocketmq/client/consumer/DefaultLitePullConsumer.java:276]: need to consider name space.

File: client/src/main/java/org/apache/rocketmq/client/consumer/DefaultLitePullConsumer.java [hunk 2]
```diff
@@ -271,6 +271,11 @@ public void assign(Collection<MessageQueue> messageQueues) {
         defaultLitePullConsumerImpl.assign(queuesWithNamespace(messageQueues));
     }
 
+    @Override
+    public void setSubExpression4Assgin(final String topic, final String subExpresion) {
```
Comment (tsunghanjacktsai) [inline client/src/main/java/org/apache/rocketmq/client/consumer/DefaultLitePullConsumer.java:275]: `setSubExpression4Assgin` -> `setSubExpression4Assign`

**Review by tsunghanjacktsai — CHANGES_REQUESTED** [review]
Hi @lizhiboo ,

Only a small mistake on the variable names and LGTM for the rest. Shall we add some tests for this?

**Review by caigy — APPROVED** [review]
LGTM

File: client/src/main/java/org/apache/rocketmq/client/consumer/DefaultLitePullConsumer.java [hunk 3]
```diff
@@ -271,6 +271,11 @@ public void assign(Collection<MessageQueue> messageQueues) {
         defaultLitePullConsumerImpl.assign(queuesWithNamespace(messageQueues));
     }
 
+    @Override
+    public void setSubExpression4Assign(final String topic, final String subExpresion) {
```
Comment (lizhanhui) [inline client/src/main/java/org/apache/rocketmq/client/consumer/DefaultLitePullConsumer.java:275]: Naming is inconsistent with the existing style

File: client/src/main/java/org/apache/rocketmq/client/consumer/LitePullConsumer.java [hunk 4]
```diff
@@ -75,6 +75,15 @@ public interface LitePullConsumer {
...
     void assign(Collection<MessageQueue> messageQueues);
 
+    /**
+     * Set topic subExpression for assign mode. This interface does not allow be call after start(). Default value is * if not set.
+     * assignment and will replace the previous assignment (if there is one).
+     *
+     * @param subExpression subscription expression.it only support or operation such as "tag1 || tag2 || tag3" <br> if
+     *      * null or * expression,meaning subscribe all
+     */
+    void setSubExpression4Assign(final String topic, final String subExpression);
```
Comment (lizhanhui) [inline client/src/main/java/org/apache/rocketmq/client/consumer/LitePullConsumer.java:85]: Same here

File: client/src/main/java/org/apache/rocketmq/client/impl/consumer/DefaultLitePullConsumerImpl.java [hunk 5]
```diff
@@ -531,6 +534,17 @@ public synchronized void assign(Collection<MessageQueue> messageQueues) {
         }
     }
 
+    public synchronized void setSubExpression4Assign(final String topic, final String subExpression) {
```
Comment (lizhanhui) [inline client/src/main/java/org/apache/rocketmq/client/impl/consumer/DefaultLitePullConsumerImpl.java:537]: Same here

File: client/src/main/java/org/apache/rocketmq/client/impl/consumer/DefaultLitePullConsumerImpl.java [hunk 6]
```diff
@@ -121,6 +122,8 @@ private enum SubscriptionType {
 
     private static final long PULL_TIME_DELAY_MILLS_ON_EXCEPTION = 3 * 1000;
 
+    private ConcurrentHashMap<String/* topic */, String/* subExpression */> topic2SubExpression = new ConcurrentHashMap<>();
```
Comment (lizhanhui) [inline client/src/main/java/org/apache/rocketmq/client/impl/consumer/DefaultLitePullConsumerImpl.java:125]: Same here

**Review by ni-ze — APPROVED** [review]
LGTM~

File: client/src/main/java/org/apache/rocketmq/client/consumer/DefaultLitePullConsumer.java (hunk 3 above)
Comment (lizhiboo) [reply client/src/main/java/org/apache/rocketmq/client/consumer/DefaultLitePullConsumer.java:275]: do u have any suggestion for this naming?

Comment (lizhanhui) [reply client/src/main/java/org/apache/rocketmq/client/consumer/DefaultLitePullConsumer.java:275]: Use full words to make it easy to follow. No non-standard, obscure, abbreviations.

**Review by duhenglucky — APPROVED** [review]

**Pull request merged**